src/main/groovy/nebula/plugin/dependencylock/utils/DependencyResolutionVerifier.groovy

```diff
@@ -0,0 +1,150 @@
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package nebula.plugin.dependencylock.utils
+
+import nebula.plugin.dependencylock.exceptions.DependencyLockException
+import org.gradle.api.BuildCancelledException
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.artifacts.ResolveException
+import org.gradle.api.execution.TaskExecutionListener
+import org.gradle.api.logging.Logger
+import org.gradle.api.logging.Logging
+import org.gradle.api.tasks.TaskState
+import org.gradle.internal.locking.LockOutOfDateException
+
+class DependencyResolutionVerifier {
+    private static final Logger LOGGER = Logging.getLogger(DependencyResolutionVerifier.class)
+    private static final String UNRESOLVED_DEPENDENCIES_FAIL_THE_BUILD = 'dependencyLock.unresolvedDependenciesFailTheBuild'
+
+    static void verifySuccessfulResolution(Project project) {
+        Boolean unresolvedDependenciesShouldFailTheBuild = project.hasProperty(UNRESOLVED_DEPENDENCIES_FAIL_THE_BUILD)
+                ? (project.property(UNRESOLVED_DEPENDENCIES_FAIL_THE_BUILD) as String).toBoolean()
+                : true
+
+        Map<String, Set<Configuration>> failedDepsByConf = new HashMap<String, Set<Configuration>>()
+        Set<String> lockedDepsOutOfDate = new HashSet<>()
+
+        def projectExecutionEnabled = project.gradle.startParameter.isParallelProjectExecutionEnabled()
+
+        project.gradle.taskGraph.whenReady { taskGraph ->
+            LinkedList tasks = taskGraph.executionPlan.executionQueue
+
+            Task lastTask
+            if (projectExecutionEnabled) {
+                lastTask = tasks
+                        .groupBy { task -> task.project }
+                        .find { proj, tasksForProj -> proj == project }
+                        .value
+                        .last()
+                        ?.task
+            } else {
+                lastTask = tasks.last?.task
+            }
+
+            taskGraph.addTaskExecutionListener(new TaskExecutionListener() {
+                @Override
+                void beforeExecute(Task task) {
+                    //DO NOTHING
+                }
+
+                @Override
+                void afterExecute(Task task, TaskState taskState) {
+                    if (task.path == lastTask.path && !taskState.failure) {
+                        project.configurations.matching { // returns a live collection
+                            (it as Configuration).state != Configuration.State.UNRESOLVED &&
```

Comment:
Does not re-resolve dependencies, but gets called only when dependencies have been resolved correctly or with errors